File: tests/cat-art.test.js
const ART = require("../cat.js");

test("phraseForSecond is deterministic and bounded", () => {
  for (let s = 0; s < 600; s++) {
    const p = ART.phraseForSecond(s);
    assert.truthy(typeof p === "string" && p.length > 0);
  }
  assert.eq(ART.phraseForSecond(0), ART.phraseForSecond(ART.PHRASES.length));
});

test("phraseForSecond handles negative seconds", () => {
  assert.eq(ART.phraseForSecond(-3), ART.phraseForSecond(3));
});

test("svgMarkup contains an svg root and the orange palette", () => {
  const svg = ART.svgMarkup();
  assert.truthy(svg.includes("<svg"));
  assert.truthy(svg.includes("</svg>"));
  // Orange-y stops we depend on for the "fat orange cat" vibe.
  assert.truthy(svg.includes("#ef8b2a"));
});

test("personality phrases stay on-brand (no apologies)", () => {
  for (const p of ART.PHRASES) {
    const lower = p.toLowerCase();
    assert.truthy(!lower.includes("sorry"), `apology in: ${p}`);
    assert.truthy(!lower.includes("please"), `politeness in: ${p}`);
  }
});

File: android-app/app/src/test/kotlin/com/fatorangecat/core/CatArtTest.kt
package com.fatorangecat.core

import org.junit.Assert.assertEquals
import org.junit.Assert.assertFalse
import org.junit.Assert.assertTrue
import org.junit.Test

class CatArtTest {

    @Test
    fun phraseForSecond_isDeterministicAndBounded() {
        for (s in 0 until 600) {
            val p = CatArt.phraseForSecond(s)
            assertTrue(p.isNotEmpty())
        }
        assertEquals(CatArt.phraseForSecond(0), CatArt.phraseForSecond(CatArt.PHRASES.size))
    }

    @Test
    fun phraseForSecond_handlesNegativeSeconds() {
        assertEquals(CatArt.phraseForSecond(3), CatArt.phraseForSecond(-3))
    }

    @Test
    fun svgMarkup_containsAnSvgRootAndTheOrangePalette() {
        val svg = CatArt.svgMarkup()
        assertTrue(svg.contains("<svg"))
        assertTrue(svg.contains("</svg>"))
        // Orange-y stops we depend on for the "fat orange cat" vibe.
        assertTrue(svg.contains("#ef8b2a"))
    }

    @Test
    fun phrases_stayOnBrand_noApologiesNoPlease() {
        for (p in CatArt.PHRASES) {
            val lower = p.lowercase()
            assertFalse("apology in: $p", lower.contains("sorry"))
            assertFalse("politeness in: $p", lower.contains("please"))
        }
    }
}
